app/src/main/java/org/mozilla/fenix/utils/ToolbarPopupWindow.kt

```diff
@@ -119,12 +121,4 @@ object ToolbarPopupWindow {
             selectedTab?.readerState?.activeUrl ?: selectedTab?.content?.url
         }
     }
-
-    private fun ClipboardHandler.getUrl(): String? {
-        if (containsURL()) {
-            text?.let { return it }
-            Logger("ToolbarPopupWindow").error("Clipboard contains URL but unable to read text")
-        }
-        return null
-    }
```

Comment:
I understand from https://github.com/mozilla-mobile/fenix/issues/27182#issuecomment-1275144901 that there are security concerns related to pasting non-url text into the awesomebar. I apologize that I don't actually have any context around them when I started reviewing this PR which I have falsely assumed was only about fixing the system notification around pasting. I am still happy to review this and get us to point where we can land this if we only limited the changes to only resolving showing the system notification. I have started pinging some people to get back to you regarding https://github.com/mozilla-mobile/fenix/issues/27182#issuecomment-1275856232, but I unfortunately do not have more context here.

Reply:
If possible, one way forward would be to move the changes related to https://github.com/mozilla-mobile/fenix/issues/27182 to its own separate PR. Sorry for any inconvenience 🙇